plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/DeployImageWizard.java

```diff
@@ -61,14 +61,26 @@ public DeployImageWizard(IDockerImage image, Connection connection, IProject pro
 			IDockerConnection dockerConnection = image.getConnection();
 			model.setOriginatedFromDockerExplorer(true);
 			model.setDockerConnection(dockerConnection);
-			model.setImageName(image.repo());
+			model.setImageName(getImageNameWithTag(image));
 		}
 		model.setStartedWithActiveConnection(isAuthorized);
 		model.initModel(connection, project, isAuthorized);
 
 		setNeedsProgressMonitor(true);
 	}
 
+	private String getImageNameWithTag(IDockerImage image) {
```

Comment:
Content provider for Docker explorer for each original image creates a set of 'explorer' images based on repoTag, so that each 'explorer' image mostly has 1 tag + occasionally 'latest'. That can be seen on the screenshot to the issue. When user selects an image in Docker explorer, handler gets image object not with all tags for the repo, but most often 1. At least I see it this way. @jcantrill could you please run this PR?